website/views.py

```diff
@@ -46,41 +68,55 @@ def home(request):
     return render(request, "website/templates/index.html", context)
 
 # to get all questions posted till now and pagination, 20 questions at a time
+
+
 def questions(request):
-    questions = Question.objects.all().filter(is_spam = False).order_by('-date_created')
+    if is_moderator(request.user) and settings.MODERATOR_ACTIVATED:
+        return HttpResponseRedirect('/moderator/questions/')
+    categories = FossCategory.objects.order_by('name')
+    questions = Question.objects.all().filter(
+        is_spam=False, is_active=True).order_by('-date_created')
     context = {
+        'categories': categories,
         'questions': questions,
     }
     return render(request, 'website/templates/questions.html', context)
 
 # get particular question, with votes,anwsers
-def get_question(request, question_id = None, pretty_url = None):
 
-    question = get_object_or_404(Question, id = question_id)
+
+def get_question(request, question_id=None, pretty_url=None):
+    if is_moderator(request.user,get_object_or_404(Question, id=question_id)) and settings.MODERATOR_ACTIVATED:
```

Comment:
use a variable for these function calls:

get_object_or_404(Question, id=question_id)
get_object_or_404(Question, id=question_id, is_active=True)